utils/vpinns/quad_rule.py

```diff
@@ -147,17 +148,25 @@ def get_test_function_properties(n_test, coord_quadrature, approach="2"):
     """
     
     dict_legendre = {"1" : modified_legendre,
-                     "2" : modified_legendre_2}
+                     "2" : modified_legendre_2,
+                     "3" : legendre_basis_shape_functions,
+                     "4" : lagrange_basis_shape_functions}
     
     dict_legendre_derivative = {"1" : modified_legendre_derivative,
-                                "2" : modified_legendre_derivative_2}
+                                "2" : modified_legendre_derivative_2,
+                                "3" : legendre_basis_shape_functions_derivative,
+                                "4": lagrange_basis_shape_functions_derivative}
```

Comment:
Same here :)

Reply:
And another idea would be to combine both dictionaries into one (nested) dictionary, to make sure that the method used for computing derivatives matches the one used to evaluate the functions (unless you explicitly want to allow that? 🤔), e.g.

```python
dict_legendre = {
    "modified_legendre_scipy" : {
        "value" : modified_legendre,
        "derivative" : modified_legendre_derivative
    },
    "modified_legendre_custom" : {
        "value" : modified_legendre_2,
        "derivative" : modified_legendre_derivative_2
    },
    # ... 
    # CONTINUE HERE
    # ...
}
```